### DISCUSSION TITLE
query_decay

### DISCUSSION THREAD

**Discussion by antoineeripret** [antoineeripret/gsc_wrapper#6] [Ideas]
Also, I don't know if anyone asked for this, but it could be really great to get the decay function to work with weeks and queries. I did get it to work with queries, I didn't have the time to refactor it for weeks, if I make it of course I'd share it... Thank you

**Comment by dtaubaso** [comment]
What I didi was simply change "query" on "page" in the groupby and other instances, obviously having a dataframe with queries

`.groupby(['query','yearMonth'], as_index=False)`

**Reply by antoineeripret** [reply]
@dtaubaso: 

Didn't you want the data per week? This is the tricky part, otherwise updating the function to use query instead of page is indeed easy. Also, in the context you mentioned, would you rather use clicks or impressions? 

Thanks.

**Reply by dtaubaso** [reply]
Yeah, I know that's the tricky part, that's why I couldn't do it! 😄 Impressions is fine!

**Reply by antoineeripret** [reply]
@dtaubaso, I've worked on a revamp for the method to make it more generic. The new structure would be the following: 

```python
.find_content_decay(
        #same as before even if name has changed 
        threshold_metric=0,
        #to choose between clicks and impressions 
        metric = 'clicks', 
       #same as before 
        threshold_decay=0.1, 
        #choose between query or page 
        type='query', 
        #to choose between week and month 
        period='month')
 )
```

The output could be something like this, where the value of **period_max** could be a Month (%Y-%m) or a week (%Y-%u)

|query|metric_last_period|metric_max|period_max|decay|decay_abs|
| -------- | ------- |------- |------- |------- |------- |
|xxxxx|39585|78269|2023-04|0.494244|38684|

Is it what you were looking for?

**Comment by dtaubaso** [comment]
This is amazing!! Thank you!!

**Comment by antoineeripret** [comment]
@dtaubaso: update uploaded. New function available!
